### PR TITLE
Use black to format notebooks

### PR DESCRIPTION
**Pull request by jsignell**
**Related Issue(s):**

- #1040 

**PR Checklist:**

- [x] Code is formatted (run `pre-commit run --all-files`)
- [x] Tests pass (run `scripts/test`)
- [x] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -11,6 +11,14 @@ repos:
         stages: [commit]
         types: [python]
 
+      - id: black-jupyter
+        name: black-jupyter
+        entry: black
+        language: python
```
Comment (gadomski) [inline .pre-commit-config.yaml:17]: I'm not quite sure of the lineage of `language: system` for the other pre-commit hooks, but we should _probably_ be consistent?

**Review by gadomski — COMMENTED** [review]
I'm assuming the notebook changes are all formatting, no functional changes (read: I didn't look at them all)?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jsignell) [reply .pre-commit-config.yaml:17]: Yeah I tried it with `system` and it didn't like that plus the additional dependencies field. I have never seen `language: system` before, but I was hesitant about changing all the other pre-commit actions.

Comment (gadomski) [reply .pre-commit-config.yaml:17]: Hmm...okay no worries. Per https://github.com/stac-utils/pystac/pull/1034#issuecomment-1468032168, I might be taking some time soon to write up some standards docs for this sort of stuff (e.g. **pystac-client** doesn't use `language: system`), which will be a chance to revisit how we set up the **pre-commit** hooks. So I think it's fine to do it the way that works now, and then we can bring everything in line later when we hit all the repos. Thanks.

**Comment by jsignell** [comment]
> I'm assuming the notebook changes are all formatting, no functional changes (read: I didn't look at them all)?

Yeah it's almost all replacing `'` with `"`

**Review by gadomski — APPROVED** [review]

**Pull request merged**